test/math/ring_signature.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <random>
+#include <boost/test/unit_test.hpp>
+#include <bitcoin/bitcoin.hpp>
+
+using namespace bc;
+
+data_chunk new_seed()
+{
+    data_chunk seed(10);
+    pseudo_random_fill(seed);
```

Comment:
Don't use dynamically-selected test vectors.

Reply:
This is what the secp256k1-zkp unit test does. i just copied it.

Reply:
I don’t expect them to follow our practices, but we don’t want unit tests that change on every execution. The fauilures would be untraceable. Generation of one random but fixed vector is sufficient if randomness is sufficient.

Reply:
OK, going to generate should standard test cases based off their unit test, translating the values to our code. Here is the code I'm referring to https://github.com/ElementsProject/secp256k1-zkp/blob/secp256k1-zkp/src/modules/rangeproof/tests_impl.h#L341